### ISSUE TITLE
Horizontal scroll bar is full and disabled when not all columns are displayed

### ISSUE THREAD

**Issue by owenca** [HaikuArchives/FontBoy#29]
See the screenshot below. There are 22 fonts, only 18 of which are displayed. You have to resize the window for the horizontal scroll bar to update.

![screen shot 2017-12-31 at 10 31 02 pm](https://user-images.githubusercontent.com/6434360/34465996-5de58cba-ee7a-11e7-90d8-683b77f84d6d.png)

**Comment by owenca** [comment]
This issue seems to only occur with the default number (22) of fonts. The window is resized to an incorrect height, leaving blank space at the bottom.

Also, if you manually resize the window, say, shrinking its height, the numbers of displayed rows will decrease prematurely, long before the horizontal scrollbar is almost touching the bottom row.

I will dig more into it if this bug bothers people enough.

**Comment by janus2** [comment]
The first time you start the application the window is not resized the size is predefined when you reopen the window the size and position are the one saved at the end of the last session. (A lot of BeOS applications use this method)

**Comment by owenca** [comment]
Yes, but the application may choose to resize the window after opening it with a predefined size or the size from the settings file. This is the case with FontBoy; it just didn't calculate the height correctly.
